### PR TITLE
Fix return type of the_title() and single_month_title()

### PR DESCRIPTION
**Pull request by IanDelMar**
This PR fixes the return types for `the_title()` and `single_month_title()` in `EchoParameterDynamicFunctionReturnTypeExtension`.

I forgot to add the other types in #111 and #143.

### CODE REVIEW

File: src/EchoParameterDynamicFunctionReturnTypeExtension.php [hunk 1]
```diff
@@ -78,13 +78,44 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
...
 
         return TypeCombinator::union(
-            new StringType(),
-            new VoidType()
+            self::getEchoFalseReturnType($name),
+            self::getEchoTrueReturnType($name)
         );
     }
+
+    protected static function getEchoTrueReturnType( string $name ): Type
```
Comment (szepeviktor) [inline src/EchoParameterDynamicFunctionReturnTypeExtension.php:91]: Why are these methods static?

File: src/EchoParameterDynamicFunctionReturnTypeExtension.php [hunk 2]
```diff
@@ -78,13 +78,44 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
...
+            return TypeCombinator::union(
+                new VoidType(),
+                new ConstantBooleanType(false)
+            );
+        }
+
+        return new VoidType();
+    }
+
+    protected static function getEchoFalseReturnType( string $name ): Type
```
Comment (szepeviktor) [inline src/EchoParameterDynamicFunctionReturnTypeExtension.php:103]:
```suggestion
    protected static function getEchoFalseReturnType(string $name): Type
```

This is not WordPress :)

File: src/EchoParameterDynamicFunctionReturnTypeExtension.php (hunk 1 above)
Comment (IanDelMar) [reply src/EchoParameterDynamicFunctionReturnTypeExtension.php:91]: Because they don't need an instantiation of the class

Comment (szepeviktor) [reply src/EchoParameterDynamicFunctionReturnTypeExtension.php:91]: Okay!

Comment (szepeviktor) [inline src/EchoParameterDynamicFunctionReturnTypeExtension.php:91]:
```suggestion
    protected static function getEchoTrueReturnType(string $name): Type
```

**Comment by szepeviktor** [comment]
@johnbillion Do you have any comments?

**Review by szepeviktor — APPROVED** [review]

**Review by johnbillion — APPROVED** [review]
Good catches.

**Pull request merged**